heureka/ui/src/components/filters/FilterSelect.jsx

```diff
@@ -46,80 +59,65 @@ const FilterSelect = ({ filters }) => {
   }
 
   const handleSearchChange = (value) => {
-    // debounce setSearchTerm to avoid unnecessary re-renders
+    // Debounce search term to avoid unnecessary re-renders
     const debouncedSearchTerm = setTimeout(() => {
       setSearchTerm(value.target.value)
     }, 500)
-
-    // clear timeout if we have a new value
     return () => clearTimeout(debouncedSearchTerm)
   }
 
-  // const handleKeyDown = (event) => {
-  //   if (event.key === "Enter") {
-  //     handleFilterValueChange()
-  //   }
-  // }
-
   return (
     <Stack alignment="center" gap="8">
       <InputGroup>
         <Select
           name="filter"
           className="filter-label-select w-64 mb-0"
           label="Filter"
-          value={filterLabel}
+          value={formatLabel(filterLabel)}
           onChange={(val) => handleFilterLabelChange(val)}
+          disabled={isLoading}
         >
           {filterLabels?.map((filter) => (
             <SelectOption
               value={filter}
-              // label={humanizeString(filter)}
-              label={filter}
+              label={formatLabel(filter)}
               key={filter}
             />
           ))}
         </Select>
-        {/* <Select
+        <Select
           name="filterValue"
           value={filterValue}
           onChange={(value) => handleFilterValueChange(value)}
-          disabled={filterLabelValues[filterLabel] ? false : true}
+          disabled={!filterLabelValues[filterLabel]?.length}
           loading={filterLabelValues[filterLabel]?.isLoading}
           className="filter-value-select w-96 bg-theme-background-lvl-0"
-          key={resetKey}
         >
-          {filterLabelValues[filterLabel]?.values
-            ?.filter(
-              (value) =>
-                // filter out already active values for this label
-                !activeFilters[filterLabel]?.includes(value)
-            )
+          {filterLabelValues[filterLabel]
+            ?.filter((value) => !activeFilters[filterLabel]?.includes(value))
```

Comment:
I would add the comment about what this does back in. It's a bit hard to read the long function chain without some context.